internal/aws/fake/client.go

```diff
@@ -16,55 +16,62 @@
 package fake
 
 import (
-	"encoding/json"
-	"fmt"
-	"sync"
 	"testing"
-	"time"
 
-	"github.com/aws/aws-lambda-go/events"
 	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/service/sqs/types"
-	"github.com/stretchr/testify/require"
-
 	"github.com/observiq/bindplane-otel-collector/internal/aws/client"
+	"github.com/observiq/bindplane-otel-collector/internal/aws/event"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.uber.org/zap"
 )
 
 var _ client.Client = &AWS{}
 
 // AWS is a fake AWS client
 type AWS struct {
+	set component.TelemetrySettings
+
 	s3Client  *s3Client
 	sqsClient *sqsClient
 
-	count   int
-	countMu sync.Mutex
+	eventMarshaler event.Marshaler
+}
+
+// ClientOption is a function that modifies the AWS client
+type ClientOption func(*AWS)
+
+// WithFDRBodyMarshaler sets the event marshaler to the FDR marshaler
+func WithFDRBodyMarshaler() ClientOption {
+	return func(f *AWS) {
+		f.eventMarshaler = event.NewFDRMarshaler(f.set)
+	}
 }
 
 // SetFakeConstructorForTest sets the fake constructor for the AWS client
 // It returns a function that restores the original constructor
 // It is intended to be used in a defer statement
 // e.g. defer fake.SetFakeConstructorForTest(t)()
-func SetFakeConstructorForTest(t *testing.T) func() {
+func SetFakeConstructorForTest(t *testing.T, opts ...ClientOption) func() {
 	realNewClient := client.NewClient
+	set := componenttest.NewNopTelemetrySettings()
+	set.Logger = zap.NewNop()
 	client.NewClient = func(_ aws.Config) client.Client {
-		return &AWS{
-			s3Client:  NewS3Client(t).(*s3Client),
-			sqsClient: NewSQSClient(t).(*sqsClient),
+		f := &AWS{
+			set:            set,
+			s3Client:       NewS3Client(t).(*s3Client),
+			sqsClient:      NewSQSClient(t).(*sqsClient),
+			eventMarshaler: event.NewS3Marshaler(set),
+		}
+		for _, opt := range opts {
+			opt(f)
 		}
+		return f
 	}
 
-	return func() {
-		client.NewClient = realNewClient
-	}
-}
-
-// NewClient creates a new fake AWS client
-func NewClient(t *testing.T) client.Client {
```

Comment:
Rather than calling this directly, tests should call `SetFakeConstructorForTest` and then use `client.NewClient`. This forces all tests to use the same path for setting up the fakes and accounts for the handful of changes in the receiver which are included in this PR.